core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -407,6 +407,24 @@ public ServerBuilder service(PathMapping pathMapping, Service<HttpRequest, HttpR
     }
 
     /**
+     * Decorates and binds the specified {@link ServiceWithPathMapping} at multiple {@link PathMapping}
+     * of the default {@link VirtualHost}.
+     *
+     * @throws IllegalStateException if the default {@link VirtualHost} has been set via
+     *                               {@link #defaultVirtualHost(VirtualHost)} already
+     */
+    public <T extends ServiceWithPathMapping<HttpRequest, HttpResponse>,
+            R extends Service<HttpRequest, HttpResponse>>
+            ServerBuilder service(T serviceWithPathMapping, Function<T, R> decorator) {
+        requireNonNull(serviceWithPathMapping, "serviceWithPathMapping");
+        requireNonNull(serviceWithPathMapping.pathMappings(), "serviceWithPathMapping.pathMappings()");
```

Comment:
If you think so, then how about just doing
<code>
if (serviceWithPathMapping.pathMappings() != null) {
    ...
}
</code>
since `requireNonNull` is designed for doing parameter validation in methods and constructor, not for the returning value from the method call.  :-)